### PR TITLE
Add new webhooks and rename legacy webhooks

### PR DESCRIPTION
**Pull request by lenkaklimcikova**
### Motivation
Enable customers to use both webhook versions - new and legacy.
DEL-4734

### CODE REVIEW

File: Kontent.Ai.Management/Models/Webhooks/Triggers/Asset/AssetActionModel.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using Newtonsoft.Json;
+
+namespace Kontent.Ai.Management.Models.Webhooks.Triggers.Asset;
+
+/// <summary>
+/// Represents asset actions.
```
Comment (Sevitas) [inline Kontent.Ai.Management/Models/Webhooks/Triggers/Asset/AssetActionModel.cs:6]: singular - > Represents asset action.

File: Kontent.Ai.Management/Models/Webhooks/Triggers/ContentItem/ContentItemActionModel.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace Kontent.Ai.Management.Models.Webhooks.Triggers.ContentItem;
+
+/// <summary>
+/// Represents content item actions.
```
Comment (Sevitas) [inline Kontent.Ai.Management/Models/Webhooks/Triggers/ContentItem/ContentItemActionModel.cs:7]: singular - > Represents content item action.

File: Kontent.Ai.Management/Models/Webhooks/Triggers/ContentType/ContentTypeActionModel.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿using Newtonsoft.Json;
+
+namespace Kontent.Ai.Management.Models.Webhooks.Triggers.ContentType;
+
+/// <summary>
+/// Represents content type actions.
```
Comment (Sevitas) [inline Kontent.Ai.Management/Models/Webhooks/Triggers/ContentType/ContentTypeActionModel.cs:6]: singular - > Represents * action.

File: Kontent.Ai.Management/Models/Webhooks/Triggers/Language/LanguageActionModel.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+﻿using Newtonsoft.Json;
+
+namespace Kontent.Ai.Management.Models.Webhooks.Triggers.Language;
+
+/// <summary>
+/// Represents language actions.
```
Comment (Sevitas) [inline Kontent.Ai.Management/Models/Webhooks/Triggers/Language/LanguageActionModel.cs:6]: singular - > Represents * action.

File: Kontent.Ai.Management/Models/Webhooks/Triggers/Taxonomy/TaxonomyActionModel.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿using Newtonsoft.Json;
+
+namespace Kontent.Ai.Management.Models.Webhooks.Triggers.Taxonomy;
+
+/// <summary>
+/// Represents taxonomy actions.
```
Comment (Sevitas) [inline Kontent.Ai.Management/Models/Webhooks/Triggers/Taxonomy/TaxonomyActionModel.cs:6]: singular - > Represents * action.

File: Kontent.Ai.Management/Models/Webhooks/Triggers/DeliverySlotModel.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
+﻿using System.Runtime.Serialization;
+
+namespace Kontent.Ai.Management.Models.Webhooks.Triggers;
+
+/// <summary>
+/// Represents the delivery slot model.
+/// </summary>
+public enum DeliverySlotModel
```
Comment (Sevitas) [inline Kontent.Ai.Management/Models/Webhooks/Triggers/DeliverySlotModel.cs:8]: DeliverySlot

File: Kontent.Ai.Management/Models/Webhooks/Triggers/DeliverySlotModel.cs [hunk 7]
```diff
@@ -0,0 +1,21 @@
+﻿using System.Runtime.Serialization;
+
+namespace Kontent.Ai.Management.Models.Webhooks.Triggers;
+
+/// <summary>
+/// Represents the delivery slot model.
```
Comment (Sevitas) [inline Kontent.Ai.Management/Models/Webhooks/Triggers/DeliverySlotModel.cs:6]: /// Represents the delivery slot.

File: Kontent.Ai.Management/Models/LegacyWebhooks/LegacyWebhookCreateModel.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
+﻿using Kontent.Ai.Management.Models.LegacyWebhooks.Triggers;
+using Newtonsoft.Json;
+
+namespace Kontent.Ai.Management.Models.LegacyWebhooks;
+
+/// <summary>
+/// Represents the webhook create model.
```
Comment (Sevitas) [inline Kontent.Ai.Management/Models/LegacyWebhooks/LegacyWebhookCreateModel.cs:7]: /// Represents the legacy webhook create model.

**Review by Sevitas — CHANGES_REQUESTED** [review]

File: Kontent.Ai.Management.Tests/ManagementClientTests/LegacyWebhookTests.cs [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+using Kontent.Ai.Management.Models.Shared;
+using Kontent.Ai.Management.Tests.Base;
+using System;
+using System.Net.Http;
+using Xunit;
+using static Kontent.Ai.Management.Tests.Base.Scenario;
+
+namespace Kontent.Ai.Management.Tests.ManagementClientTests;
+
+public class LegacyWebhookTests : IClassFixture<FileSystemFixture>
```
[the remark this answers is not in the document]
Comment (lenkaklimcikova) [reply Kontent.Ai.Management.Tests/ManagementClientTests/LegacyWebhookTests.cs:13]: fixed

File: Kontent.Ai.Management/Models/LegacyWebhooks/Triggers/LegacyWebhookTriggersModel.cs
[the remark this answers is not in the document]
Comment (lenkaklimcikova) [reply Kontent.Ai.Management/Models/LegacyWebhooks/Triggers/LegacyWebhookTriggersModel.cs:1]: good point thanks, didn't think about changing the old ones,too :D fixed

File: Kontent.Ai.Management/Models/Webhooks/Triggers/WebhookEventsFilterModel.cs [hunk 10]
```diff
@@ -0,0 +1,21 @@
+﻿using System.Runtime.Serialization;
+
+namespace Kontent.Ai.Management.Models.Webhooks.Triggers;
+
+/// <summary>
+/// Represents the webhook events filter model.
+/// </summary>
+public enum WebhookEventsFilterModel
```
[the remark this answers is not in the document]
Comment (lenkaklimcikova) [reply Kontent.Ai.Management/Models/Webhooks/Triggers/WebhookEventsFilterModel.cs:8]: fixed

File: Kontent.Ai.Management/Models/Webhooks/WebhookModel.cs [hunk 11]
```diff
@@ -9,46 +9,46 @@ namespace Kontent.Ai.Management.Models.Webhooks;
...
-
+    
+    /// <summary>
+    /// Gets or sets ISO-8601 formatted date/time of the last change to the webhook.
+    /// </summary>
+    [JsonProperty("last_modified")]
+    public DateTime? LastModified { get; set; }
+    
     /// <summary>
     /// Gets or sets the specific events that trigger the webhook. At least one valid trigger is required.
```
[the remark this answers is not in the document]
Comment (lenkaklimcikova) [reply Kontent.Ai.Management/Models/Webhooks/WebhookModel.cs:50]: I did it with an example in the legacy webhooks, it was not there either. Should I add it into the LegacyWebhookModel as well then? :)

File: Kontent.Ai.Management.Tests/ManagementClientTests/LegacyWebhookTests.cs [hunk 12]
```diff
@@ -0,0 +1,238 @@
...
+        _scenario
+            .CreateExpectations()
+            .HttpMethod(HttpMethod.Get)
+            .Response(response)
+            .Url($"{Endpoint}/projects/{PROJECT_ID}/webhooks/{identifier.Id}")
+            .Validate();
+    }
+
+    [Fact]
+    public async void GetWebhookAsync_ByCodename_GetsWebhook()
```
[the remark this answers is not in the document]
Comment (lenkaklimcikova) [reply Kontent.Ai.Management.Tests/ManagementClientTests/LegacyWebhookTests.cs:58]: Yeah, those were wrongly written before, I fixed it in new tests but forgot to fix the old ones.

**Comment by Sevitas** [comment]
You have renamed the action enums to names with suffix Enum, please keep it consistent in the namespace _Kontent.Ai.Management.Models.Webhooks_. There are enums without this suffix. I'd prefer removing those suffixes

File: Kontent.Ai.Management/Models/Webhooks/WebhookModel.cs (hunk 11 above)
[the remark this answers is not in the document]
Comment (Sevitas) [reply Kontent.Ai.Management/Models/Webhooks/WebhookModel.cs]: Please add it for LegacyWebhookModel as well.

[the remark this answers is not in the document]
Comment (lenkaklimcikova) [reply Kontent.Ai.Management/Models/Webhooks/WebhookModel.cs]: added

**Review by Sevitas — APPROVED** [review]

File: Kontent.Ai.Management/Models/Webhooks/Triggers/Language/LanguageActionModel.cs (hunk 4 above)
Comment (pokornyd) [reply Kontent.Ai.Management/Models/Webhooks/Triggers/Language/LanguageActionModel.cs]: fixed typo

**Review by pokornyd — DISMISSED** [review]
following internal discussion, the target framework was updated to .NET 8.0 for all projects under the solution.

**Review by pokornyd — APPROVED** [review]

**Pull request merged**